HST/product_labels/__init__.py

```
@@ -651,40 +656,37 @@ def find_suffix_dicts(ipppssoot_dict, suffixes):
+            if reference_suffixes:
+                break
 
         ipppssoot_dict['reference_suffixes'] = set(reference_suffixes)
         ipppssoot_dict['reference_dicts'] = reference_dicts
 
+        # Log the reference found
         count = len(reference_dicts)
-        if count > 1:
+        if count > 1:       # If there are multiple suitable references, just pick one
             for k, reference_dict in enumerate(reference_dicts):
                 logger.info(f'Multiple {tag.lower()} files found for {ipppssoot} ' +
```

Pylint (4.0.2)

[warning] Using possibly undefined loop variable 'tag'

(W0631)